cray_product_catalog/catalog_delete.py

```diff
@@ -167,15 +185,43 @@ def main():
     PRODUCT_VERSION = os.environ.get("PRODUCT_VERSION").strip()  # required
     CONFIG_MAP = os.environ.get("CONFIG_MAP", "cray-product-catalog").strip()
     CONFIG_MAP_NS = os.environ.get("CONFIG_MAP_NAMESPACE", "services").strip()
+    PRODUCT_CONFIG_MAP = format_product_cm_name(CONFIG_MAP, PRODUCT)
     KEY = os.environ.get("KEY", "").strip() or None
+    MAX_RETRIES = 100
+    MAX_RETRIES_FOR_PROD_CM = 10
 
-    args = (CONFIG_MAP, CONFIG_MAP_NS, PRODUCT, PRODUCT_VERSION, KEY)
-    LOGGER.info(
-        "Removing from ConfigMap=%s in namespace=%s for %s/%s (key=%s)",
-        *args
-    )
     load_k8s()
-    modify_config_map(*args)
+
+    if KEY:
+        if KEY in CONFIG_MAP_FIELDS:
+            args = (CONFIG_MAP, CONFIG_MAP_NS, PRODUCT, PRODUCT_VERSION, MAX_RETRIES, KEY)
+        elif KEY in PRODUCT_CM_FIELDS:
+            args = (PRODUCT_CONFIG_MAP, CONFIG_MAP_NS, PRODUCT, PRODUCT_VERSION, MAX_RETRIES_FOR_PROD_CM, KEY)
+        else:
+            LOGGER.error(
+                "Invalid KEY=%s is input so exiting...",
+                KEY
+            )
+            return
+        LOGGER.info(
+            "Removing from config_map=%s in namespace=%s for %s/%s (key=%s)",
+            *args
+        )
+        modify_config_map(*args)
+    else:
+        args = (CONFIG_MAP, CONFIG_MAP_NS, PRODUCT, PRODUCT_VERSION, MAX_RETRIES, KEY)
+        LOGGER.info(
+            "Removing from config_map=%s in namespace=%s for %s/%s (key=%s)",
+            *args
+        )
+        modify_config_map(*args)
+
+        args = (PRODUCT_CONFIG_MAP, CONFIG_MAP_NS, PRODUCT, PRODUCT_VERSION, MAX_RETRIES_FOR_PROD_CM, KEY)
```

Comment:
If you made the earlier change to give a default value for max_attempts and move it to being the final function argument, then of course these argument lists also need to be changed:
```suggestion
            args = (CONFIG_MAP, CONFIG_MAP_NS, PRODUCT, PRODUCT_VERSION, KEY, MAX_RETRIES)
        elif KEY in PRODUCT_CM_FIELDS:
            args = (PRODUCT_CONFIG_MAP, CONFIG_MAP_NS, PRODUCT, PRODUCT_VERSION, KEY, MAX_RETRIES_FOR_PROD_CM)
        else:
            LOGGER.error(
                "Invalid KEY=%s is input so exiting...",
                KEY
            )
            return
        LOGGER.info(
            "Removing from config_map=%s in namespace=%s for %s/%s (key=%s)",
            *args
        )
        modify_config_map(*args)
    else:
        args = (CONFIG_MAP, CONFIG_MAP_NS, PRODUCT, PRODUCT_VERSION, KEY, MAX_RETRIES)
        LOGGER.info(
            "Removing from config_map=%s in namespace=%s for %s/%s (key=%s)",
            *args
        )
        modify_config_map(*args)

        args = (PRODUCT_CONFIG_MAP, CONFIG_MAP_NS, PRODUCT, PRODUCT_VERSION, KEY, MAX_RETRIES_FOR_PROD_CM)
```